### PR TITLE
Throw error on division by 0 for rationals

### PR DESCRIPTION
**Pull request by Geometer1729**
I noticed down stream that rationals weren't throwing an error on division by 0 like I expected, so I added some tests and fixed it so they do.

### CODE REVIEW

File: plutarch-test/src/Plutarch/RationalSpec.hs [hunk 1]
```diff
@@ -62,3 +62,7 @@ spec = do
         it "0.5" $ passert $ (0.5 :: Term s PRational) #== pfromData (pdata 0.5)
         it "2" $ passert $ (2 :: Term s PRational) #== pfromData (pdata 2)
         it "11/3" $ passert $ (11 / 3 :: Term s PRational) #== pfromData (pdata $ 11 / 3)
+    describe "div by 0 tests" $ do
```
Comment (srid) [inline plutarch-test/src/Plutarch/RationalSpec.hs:65]: nit: the " tests" suffix is redundant.

**Review by srid — DISMISSED** [review]

**Review by srid — APPROVED** [review]

**Comment by L-as** [comment]
CI hasn't run

**Comment by L-as** [comment]
Goldens need to be updated

**Review by L-as — CHANGES_REQUESTED** [review]
See above

File: Plutarch/Rational.hs [hunk 2]
```diff
@@ -177,21 +178,31 @@ instance Fractional (Term s PRational) where
...
+      x
+
 preduce :: Term s (PRational :--> PRational)
 preduce = phoistAcyclic $
   plam $ \x ->
     pmatch x $ \(PRational xn xd) ->
       plet (pgcd # xn # xd) $ \r ->
         plet (signum xd) $ \s ->
-          pcon $ PRational (s * pdiv # xn # r) (s * pdiv # xd # r)
+          (pcon $ PRational (s * pdiv # xn # r) (s * pdiv # xd # r))
```
Comment (L-as) [inline Plutarch/Rational.hs:205]: Unnecessary parentheses?

File: plutarch-test/goldens/rational.ops.bench.golden [hunk 3]
```diff
@@ -1,3 +1,3 @@
-+ {"exBudgetCPU":29624522,"exBudgetMemory":70391,"scriptSizeBytes":309}
-- {"exBudgetCPU":29624522,"exBudgetMemory":70391,"scriptSizeBytes":309}
-* {"exBudgetCPU":53544398,"exBudgetMemory":123567,"scriptSizeBytes":353}
\ No newline at end of file
++ {"exBudgetCPU":31240045,"exBudgetMemory":74895,"scriptSizeBytes":330}
+- {"exBudgetCPU":31240045,"exBudgetMemory":74895,"scriptSizeBytes":330}
+* {"exBudgetCPU":55159921,"exBudgetMemory":128071,"scriptSizeBytes":374}
```
Comment (L-as) [inline plutarch-test/goldens/rational.ops.bench.golden:3]: This is quite unfortunate

**Review by L-as — CHANGES_REQUESTED** [review]

**Comment by TotallyNotChase** [comment]
continued in #427

**Pull request closed without merging**